### PR TITLE
MapView to support MKPointAnnotation using new attribute annotate in Map...

### PR DESCRIPTION
**Pull request by ginamdar**
### MapView to support Pin annotation

    var pinLocation1 = {
			latitude: property.latitude,
			longitude: property.longitude,
			title: "I'm here.."
		};
		var pinLocation2 = {
			latitude: property.latitude + 0.1,
			longitude: property.longitude,
			title: "I'm here too!"
		};
    this.state = {propertyPoints: [pinLocation1, pinLocation2]};
    <MapView style={styles.map} region={this.state.region} annotations={this.state.propertyPoints}>
    </MapView>
mapview-pinannotations

### CODE REVIEW

**Comment by ginamdar** [comment]
Please review the changes for supporting annotation inside MapView

File: React/Views/RCTMapManager.m [hunk 1]
```diff
@@ -72,7 +72,23 @@ - (UIView *)view
...
 
+    if (!CLLocationCoordinate2DIsValid(coordinate)){
+        return;
+    }
+    MKPointAnnotation *pin = [[MKPointAnnotation alloc]init];
+    pin.coordinate = coordinate;
+    if (title.length){
+        pin.title = title;
+    }
+    [view addAnnotation:pin];
```
Comment (frantic) [inline React/Views/RCTMapManager.m]: This is never going to remove the pin, right?

**Comment by ginamdar** [comment]
Yes correct

**Comment by nicklockwood** [comment]
Wouldn't it make more sense to provide an array of pins?

Also, we'd really want to be able to remove them. We could do that by maintaining a dictionary of annotations keyed by coordinate internally and when the array is updated we'd remove the ones that are no longer in the array.

**Comment by nicklockwood** [comment]
libuicore is no longer required. If you wouldn't mind removing it, I'll get this merged.

**Comment by dvdhsu** [comment]
Running UIExplorer, going into MapView, and tapping on the annotation freezes the app. I've tried this with a blank project with MapView, along with an ordinary annotation; same problem.

Occurs in both simulator, and on device. Inspecting memory usage via Xcode reveals that after annotation is tapped, memory usage keeps on rising, and CPU suddenly jumps to ~180%, and stays around there.

screenshot 2015-04-15 20 39 03

**Comment by nicklockwood** [comment]
That's weird … I'm seeing this on iOS 8.1 simulator, but not pn iOS 8.3

**Comment by nicklockwood** [comment]
OK, I've found the problem. I'm landing a fix ASAP, but in the meantime if it's blocking you, you can fix it by replacing the layoutSubviews method in RCTMap.m with

    - (void)reactSetFrame:(CGRect)frame
    {
      self.frame = frame;
    }
    
    - (void)layoutSubviews
    {
      [super layoutSubviews];
    
      if (_legalLabel) {
        dispatch_async(dispatch_get_main_queue(), ^{
          CGRect frame = _legalLabel.frame;
          if (_legalLabelInsets.left) {
            frame.origin.x = _legalLabelInsets.left;
          } else if (_legalLabelInsets.right) {
            frame.origin.x = self.frame.size.width - _legalLabelInsets.right - frame.size.width;
          }
          if (_legalLabelInsets.top) {
            frame.origin.y = _legalLabelInsets.top;
          } else if (_legalLabelInsets.bottom) {
            frame.origin.y = self.frame.size.height - _legalLabelInsets.bottom - frame.size.height;
          }
          _legalLabel.frame = frame;
        });
      }
    }

**Comment by matthewga0** [comment]
Why shouldn't the annotation be a `node`? 

At least provide a way to customize the annotation, replace the default drop pin on iOS, leverage the appearance between iOS and Android.

**Pull request closed without merging**